pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -310,7 +313,14 @@ def handle_given_plugins(self) -> None:
                         "You can only specify one precision and one training type plugin."
                         f" Found more than 1 precision plugin: {type(plug).__name__}"
                     )
-
+            elif isinstance(plug, CheckpointIOPlugin):
+                if checkpoint is None:
+                    checkpoint = plug
+                else:
+                    raise MisconfigurationException(
+                        "You can only specify one checkpoint plugin and one training type plugin."
+                        f" Found more than 1 checkpoint plugin: {type(plug).__name__}"
```

Comment:
I see you followed the pattern of the two error messages above, but why does this need to mention the training type plugin?

```suggestion
                        "You can only specify one checkpoint plugin."
                        f" Available: {type(checkpoint).__name__}, given: {type(plug).__name__}"
```

Worth updating the messages above too...